unstructured/partition/docx.py

```diff
@@ -520,44 +519,39 @@ def _convert_table_to_html(self, table: DocxTable, is_nested: bool = False) -> s
         def iter_cell_block_items(cell: _Cell) -> Iterator[str]:
             """Generate the text of each paragraph or table in `cell` as a separate string.
 
-            A table nested in `cell` is converted to HTML and emitted as that string.
+            A table nested in `cell` is converted to the normalized text it contains.
             """
             for block_item in cell.iter_inner_content():
-                if isinstance(block_item, Paragraph):
+                if isinstance(paragraph := block_item, Paragraph):
                     # -- all docx content is ultimately in a paragraph; a nested table contributes
                     # -- structure only
-                    yield f"{html.escape(block_item.text)}"
-                elif isinstance(  # pyright: ignore[reportUnnecessaryIsInstance]
-                    block_item, DocxTable
-                ):
-                    yield self._convert_table_to_html(block_item, is_nested=True)
+                    # yield f"{html.escape(block_item.text)}"
+                    yield paragraph.text
+                elif isinstance(table := block_item, DocxTable):
+                    for row in table.rows:
+                        yield from iter_row_cells_as_text(row)
 
         def iter_row_cells_as_text(row: _Row) -> Iterator[str]:
-            """Generate the text of each cell in `row` as a separate string.
+            """Generate the normalized text of each cell in `row` as a separate string.
 
-            The text of each paragraph within a cell is separated from the next by a newline
-            (`"\n"`). A table nested in a cell is first converted to HTML and then included as a
-            string, also separated by a newline.
+            The text of each paragraph within a cell is not separated. A table nested in a cell is
+            converted to to a normalized string of its contents and combined with the text of the
+            cell that contains the table.
             """
-            # -- each omitted cell at the start of the row (pretty rare) gets the empty string --
+            # -- Each omitted cell at the start of the row (pretty rare) gets the empty string.
+            # -- This preserves column alignment when one or more initial cells are omitted.
             for _ in range(row.grid_cols_before):
                 yield ""
 
             for cell in row.cells:
-                yield "\n".join(iter_cell_block_items(cell))
+                cell_text = " ".join(iter_cell_block_items(cell))
+                yield " ".join(cell_text.split())
 
-            # -- each omitted cell at the end of the row (also rare) gets the empty string --
+            # -- Each omitted cell at the end of the row (also rare) gets the empty string. --
             for _ in range(row.grid_cols_after):
                 yield ""
```

Comment:
Do we need to cover the scenario where there is a row with a gap between cells of data? It seems like this just handles if there are cells before and/or after the data, but not in between. I'm not totally clear on what these scenarios are as it is, so disregard if it's not applicable. 
Also, just wanted to double-check that these paragraphs and tables should be joined with " " and not "\n"

Reply:
There is no such thing as an omitted cell in the middle of a row, not in DOCX at least. Even omitted cells at the beginning and end is weird and DOCX is the only format that supports that as far as I know.

There can be _empty_ cells in the middle of a row, and those resolve to the empty string. But they still occupy a position in the matrix.

---

We _could_ separate blocks within a cell with `"\n"`, but that would be reduced to " " in the HTML during the process of whitespace normalization. There's no such thing as a `"\n"` in HTML as far as semantic content is concerned; whitespace between words is whitespace and is all resolved to a single space when the HTML is rendered.

What we _could_ do is place a `<br/>` element between paragraphs, or a `<br/><br/>`, but there would be no corresponding line breaks in the text representation of the cell so the two would be inconsistent and there are other problems as well I'd have to think to remember.

This is just a reflection of a broader conundrum with tables and representing a table as a single element.

Non-Table elements are each a paragraph. So there is no occurrence of "multiple paragraphs per element" for other elements. And how we convert a table, which is inherently multiple paragraphs, into a textual representation hasn't been carefully specified and has organically developed into something of a mixed bag.

In this and following commits I make this representation at least consistent, choosing the simplest thing that could possibly work as the baseline (clean-concatenated text (CCT)).

At some point perhaps the ML or SE folks will weigh in with a point-of-view on how that can be elaborated for better recognition or recall or whatever. At that point we could take another look and refine this.

Reply:
> At some point perhaps the ML or SE folks will weigh in with a point-of-view on how that can be elaborated for better recognition or recall or whatever. At that point we could take another look and refine this.

Do we want to get feedback on this now?